src/remarx/sentence/corpus/text_input.py

```diff
@@ -0,0 +1,68 @@
+"Base text input class with common functionality"
+
+import pathlib
+from collections.abc import Generator
+from dataclasses import dataclass
+from functools import cached_property
+
+from remarx.sentence.segment import segment_text
+
+
+@dataclass
+class TextInput:
+    """Base class for file input for sentence corpus creation"""
+
+    input_file: pathlib.Path
+    "Reference to input file. Source of content for sentences."
+
+    field_names: tuple[str] = ("file", "offset", "text")
+    "List of field names for sentences from text input files."
+
+    @cached_property
+    def file_name(self) -> str:
+        """
+        Input file name. Associated with sentences in generated corpus.
+        """
+        return self.input_file.name
+
+    def get_text(self) -> Generator[dict[str, str]]:
+        """
+        Get plain-text contents for this file with any desired chunking (e.g.
+        pages or other semantic unit).
+        Default implementation does no chunking, no additional metadata.
+
+        :returns: Generator with a dictionary of text and any other metadata
+        that applies to this unit of text.
+        """
+        yield {"text": self.input_file.read_text(encoding="utf-8")}
+
+    def get_sentences(self) -> Generator[dict[str, any]]:
```

Comment:
Need to use `Any` from the `typing` module
```suggestion
    def get_sentences(self) -> Generator[dict[str, Any]]:
```

Reply:
Hmm... I don't think anything would catch it right now since we're not validating type declarations just requiring them to exist. But I'll quickly check if ruff has any small rules we might add.